static/app/utils/profiling/guards/profile.tsx

```diff
@@ -19,15 +19,30 @@ export function isJSProfile(profile: any): profile is JSSelfProfiling.Trace {
   return !('type' in profile) && Array.isArray(profile.resources);
 }
 
-export function isChromeTraceFormat(input: any): input is ChromeTrace.ProfileType {
-  return isChromeTraceArrayFormat(input) || isChromeTraceObjectFormat(input);
-}
-
 export function isChromeTraceObjectFormat(input: any): input is ChromeTrace.ObjectFormat {
   return typeof input === 'object' && 'traceEvents' in input;
 }
 
-export function isChromeTraceArrayFormat(input: any): input is ChromeTrace.ArrayFormat {
-  // @TODO we need to check if the profile actually includes the v8 profile nodes.
-  return Array.isArray(input);
+export function isChromeTraceArrayFormat(input: any): input is ChromeTrace.ProfileType {
+  return (
+    Array.isArray(input) && !!input.find(p => p.ph === 'P' && p.name === 'ProfileChunk')
+  );
+}
+
+// Typescript uses only a subset of the event types, so we need to
+// inspect the contents of the trace to determine the type of the profile.
+export function isTypescriptChromeTraceArrayFormat(
+  input: any
+): input is ChromeTrace.ArrayFormat {
+  return (
+    Array.isArray(input) && !input.some(p => p.ph === 'P' && p.name === 'ProfileChunk')
```

Comment:
A comment/reference explaining why this `P` + `ProfileChunk` indicates it's not a typescript chrome trace would be helpful.

Reply:
I left it just above on L32-34
```
// Typescript uses only a subset of the event types, so we need to
// inspect the contents of the trace to determine the type of the profile.
```
I'll add why we check for profileChunk

Reply:
as discussed on slack, a mention that we check that there are no profile chunks because there can be meta data events so a check for just `B` and `E` types would be too strict would be helpful in the comment